dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/JspOptions.java

```diff
@@ -231,9 +231,29 @@ else if (useJikesCompiler.equalsIgnoreCase("false"))
                 logger.logp(Level.INFO, CLASS_NAME, "populateOptions", "Invalid value for jdkSourceLevel = " + rawJdkSourceLevel + ".");
             }
         }
-        if (useJdkSourceLevel >= 13) {
-            logger.logp(Level.INFO, CLASS_NAME, "populateOptions", JspMessages.getMessage("jsp.jdksourcelevel.value", new Object[] { useJdkSourceLevel })); //152472
+
+        if (rawJavaSourceLevel != null && (rawJavaSourceLevel.equals("8") || rawJavaSourceLevel.equals("1.8"))) // only for Java 8 do we accept 1.8 and 8
+            rawJavaSourceLevel =  org.eclipse.jdt.internal.compiler.impl.CompilerOptions.VERSION_1_8; 
+        try {
+            if (rawJavaSourceLevel != null)
+                useJavaSourceLevel = Integer.parseInt(rawJavaSourceLevel);            
+        } catch(NumberFormatException e) {
+            if (com.ibm.ejs.ras.TraceComponent.isAnyTracingEnabled() && logger.isLoggable(Level.INFO)) {
+                logger.logp(Level.INFO, CLASS_NAME, "populateOptions", "Invalid value for javaSourceLevel = " + rawJavaSourceLevel + ".");
```

Comment:
Why is this message not translated? It does show up in the messages/console log right?

Reply:
Yes, it should appear in one of those logs. Nothing else is translated in this JspOptions. I would rather not translate just this one message, but rather create a separate issue / PR for translations for JspOptions.

Reply:
Actually, I just tested it out and I only saw them appear in the trace: 
```
[9/14/23, 10:15:16:588 EDT] 0000002f id=00000000 com.ibm.ws.jsp.JspOptions                                    > populateOptions ENTRY {keepgenerated=badvalue, usePageTagPool=true}
[9/14/23, 10:15:16:588 EDT] 0000002f id=00000000 com.ibm.ws.jsp.JspOptions                                    I populateOptions Invalid value for keepGenerated = badvalue
```

Reply:
The testing above was via the ibm-web-ext.xml file. I also tested via server.xml jspEngine and encountered the following warnings : 
```
[WARNING ] CWWKG0011W: The configuration validation did not succeed. CWWKG0081E: The value badValue for boolean attribute keepGenerated is invalid. Valid values are "true" and "false". The default value of false will be used.
[WARNING ] CWWKG0083W: A validation failure occurred while processing the [keepGenerated] property, value = [badValue]. Default value in use: [false]. 
```